File: paris2024-Back/src/main/java/com/example/paris2024Back/dtos/UserDTO.java
package com.example.paris2024Back.dtos;

import com.example.paris2024Back.domains.Ticket;
import com.example.paris2024Back.enums.UserRole;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class UserDTO {
    @NotNull(message = "Id is required")
    private Long id;


    private String name;

    @NotBlank(message = "Username is required")
    private String userName;

    @NotBlank(message = "private is required")
    private String password;

    @NotBlank(message = "Role is required")
    private UserRole userRole;


    private List<Ticket> ownedTickets;
}

File: paris2024-Front/src/app/interfaces/userDTO.ts
import { TicketDTO } from './ticketDTO';

export interface UserDTO {
  id: number;
  name: string;
  username: string;
  password: string;
  userRole: userRole;
  ownedTickets: TicketDTO[];
}
export enum userRole {
  USER = 'USER',
  Admin = 'ADMIN',
}
